lms/services/d2l_api/client.py

```diff
@@ -115,9 +116,11 @@ def group_set_groups(self, org_unit, group_category_id, user_id=None):
 
     def list_files(self, org_unit) -> List[dict]:
         """Get a nested list of files and folders for the given `org_unit`."""
-
         modules = self._get_course_modules(org_unit)
-        return list(self._find_files(org_unit, modules))
+        files = list(self._find_files(org_unit, modules))
+        self._file_service.upsert(list(self._files_for_storage(org_unit, files)))
```

Comment:
I was checking, but both listifications here are required, as these inputs get iterated multiple times, and would otherwise get consumed as a generator.